### PR TITLE
version update: github.com/moby/sys/mountinfo from v0.6.2 -> v0.7.1

### PR DESCRIPTION
**Pull request by ritster**
We are currently using v0.6.2 of the module github.com/moby/sys/mountinfo, we should switch to v0.7.1; the new version introduces fixes for a couple of bugs, which are discussed here:
- https://github.com/moby/sys/pull/128
- https://github.com/moby/sys/pull/135

/sig node
/sig storage

#### What type of PR is this?

/kind bug

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
@ritster i tried to fix the conflicts and that did not work. please rebase against master and push your PR again. thanks!

**Comment by ritster** [comment]
> @ritster i tried to fix the conflicts and that did not work. please rebase against master and push your PR again. thanks!

Thanks @dims. I rebased against master and pushed, please take a look!

**Comment by dims** [comment]
Thanks @ritster looks like some focused changes in one file worth picking up.

/approve
/lgtm

File: vendor/github.com/moby/sys/mountinfo/mountinfo_linux.go [hunk 1]
```diff
@@ -127,8 +131,40 @@ func GetMountsFromReader(r io.Reader, filter FilterFunc) ([]*Info, error) {
...
+	// that the thread referenced by /proc/thread-self stays alive until we
+	// finish parsing the file.
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	var f *os.File
+	if haveProcThreadSelf {
+		f, err = os.Open("/proc/thread-self/mountinfo")
+	} else {
+		// On pre-3.17 kernels (such as CentOS 7), we don't have
```
Comment (dims) [inline vendor/github.com/moby/sys/mountinfo/mountinfo_linux.go:155]: looking at the minimums for kernel versions, we don't really hit this code path:
- https://github.com/kubernetes/kubernetes/blob/7060e485694e82698c050a6a6c6ae2af20cf5e57/pkg/util/kernel/constants.go#L45

Longer discussion still ongoing points to ` probably 4.18/4.19+` as minimum:
- https://github.com/kubernetes/kubernetes/issues/116799

**Pull request merged**